form.js

```diff
@@ -0,0 +1,296 @@
+			break;
+
+		case "email":
+			everyGoodPath(emailSyntax);
+			break;
+	}
+};
+
+const sendError = (whatKindOfInput, error) => {
+	switch (whatKindOfInput) {
+		case "name":
+			everyError(nameSyntax, error);
+			break;
+
+		case "password":
+			everyError(passwordSyntax, error);
+			break;
+
+		case "ComparingPasswords":
+			everyError(ComparingPasswordsSyntax, error);
+			break;
+
+		case "city":
+			everyError(citySyntax, error);
+			break;
+
+		case "email":
+			everyError(emailSyntax, error);
+			break;
+	}
+};
+
+const checkUserName = () => {
+	if (!RegEx.CHECKING_SPECIAL_CHARACTER.test(nameSyntax.valueInput.value)) {
+		if (
+			nameSyntax.valueInput.value.length >
+			theMinimumNumberOfCharactersInTheNickname
+		) {
+			sendGoodPath("name");
+		} else {
+			sendError(
+				"name",
+				everyErrorText.THE_NAME_MUST_CONTAIN_AT_LEAST_3_LETTERS
+			);
+		}
+	} else {
+		sendError("name", everyErrorText.FORBIDDEN_SPECIAL_CHARACTERS);
+	}
+};
+
+const checkPassword = () => {
+	if (
+		passwordSyntax.valueInput.value.length >=
+		theMinimumNumberOfCharactersInThePassword
+	) {
+		if (
+			RegEx.CHECKING_SPECIAL_CHARACTER.test(passwordSyntax.valueInput.value)
+		) {
+			if (RegEx.CHECKING_NUMBER.test(passwordSyntax.valueInput.value)) {
+				sendGoodPath("password");
+			} else {
+				sendError("password", everyErrorText.ENTER_THE_NUMBER);
+			}
+		} else {
+			sendError("password", everyErrorText.SPECIAL_CHARACTERS);
+		}
+	} else {
+		sendError("password", everyErrorText.PASSWORD_IS_TOO_SHORT);
+	}
+};
+
+const checkComparingPasswords = () => {
+	if (
+		ComparingPasswordsSyntax.valueInput.value === password.value &&
+		ComparingPasswordsSyntax.valueInput.value.length > 0
+	) {
+		sendGoodPath("ComparingPasswords");
+	} else {
+		sendError("ComparingPasswords", everyErrorText.PASSWORD_ARE_NOT_THE_SAME);
+	}
+};
+
+const checkCity = () => {
+	const URL =
+		APIweather.API_LINK +
+		citySyntax.valueInput.value +
+		APIweather.API_KEY +
+		APIweather.API_UNITS;
+	axios
+		.get(URL)
+		.then(res => {
+			setCity(res.data.name);
+			sendGoodPath("city");
+			localStorage.setItem("problemWithCodeOrServer", false);
+		})
+		.catch(res => {
+			if (
+				res.response.status === APIerrors.BAD_REQUEST ||
+				res.response.status === APIerrors.NOT_FOUND
+			) {
+				return sendError("city", everyErrorText.INCORRECT_CITY_NAME);
+			}
+		})
+		.catch(() => {
+			localStorage.setItem("problemWithCodeOrServer", true);
+			citySyntax.InputIsCorrect = true;
+			allInputIsCorrect();
+			problemWithServerOrCode.style.display = "none";
+		});
+};
+
+const UpperCaseFirstLetter = word => {
+	return word.charAt(0).toUpperCase() + word.slice(1);
+};
+
+const setCity = city => {
+	localStorage.setItem("city", city);
+};
+
+const setNickName = () => {
+	const nickName = UpperCaseFirstLetter(nameSyntax.valueInput.value);
+	localStorage.setItem("nickName", `Hello ${nickName}`);
+	endFormText.innerHTML = `Hello <b>${nickName}</b>, thanks for short form`;
```

Comment:
if setCity saves item in local storage and setNickName saves item in local storage + 2 more things, setNickName should have different name or (better) be splitted into 2 functions